specification/storagecache/resource-manager/Microsoft.StorageCache/preview/2023-11-01-preview/amlfilesystem.json

```diff
@@ -0,0 +1,1191 @@
+          },
+          "type": "string"
+        },
+        "statusCode": {
+          "description": "Server-defined error code for the AML file system health",
+          "type": "string"
+        },
+        "statusDescription": {
+          "description": "Describes the health state.",
+          "type": "string"
+        }
+      },
+      "type": "object"
+    },
+    "AmlFilesystemEncryptionSettings": {
+      "description": "AML file system encryption settings.",
+      "properties": {
+        "keyEncryptionKey": {
+          "$ref": "storagecache.json#/definitions/KeyVaultKeyReference",
+          "description": "Specifies the location of the encryption key in Key Vault."
+        }
+      },
+      "type": "object"
+    },
+    "AmlFilesystemHsmSettings": {
+      "description": "AML file system HSM settings.",
+      "x-ms-mutability": [
+        "read",
+        "create"
+      ],
+      "type": "object",
+      "properties": {
+        "container": {
+          "description": "Resource ID of storage container used for hydrating the namespace and archiving from the namespace. The resource provider must have permission to create SAS tokens on the storage account.",
+          "$ref": "storagecache.json#/definitions/URLString"
+        },
+        "loggingContainer": {
+          "description": "Resource ID of storage container used for logging events and errors.  Must be a separate container in the same storage account as the hydration and archive container. The resource provider must have permission to create SAS tokens on the storage account.",
+          "$ref": "storagecache.json#/definitions/URLString"
+        },
+        "importPrefix": {
+          "description": "Only blobs in the non-logging container that start with this path/prefix get hydrated into the cluster namespace.",
+          "type": "string",
+          "default": "/"
+        }
+      },
+      "required": [
+        "container",
+        "loggingContainer"
+      ]
+    },
+    "AmlFilesystemRootSquashSettings": {
+      "description": "AML file system squash settings.",
+      "properties": {
+        "mode": {
+          "type": "string",
+          "description": "Squash mode of the AML file system.",
+          "enum": [
+            "None",
+            "RootOnly",
+            "All"
+          ],
+          "x-ms-enum": {
+            "name": "AmlFilesystemSquashMode",
+            "modelAsString": true
+          }
+        },
+        "noSquashNidLists": {
+          "description": "Semicolon separated NID IP Address list(s) to be added to the TrustedSystems.",
+          "type": "string"
+        },
+        "squashUID": {
+          "type": "integer",
+          "description": "User ID to squash to.",
+          "format": "int64"
+        },
+        "squashGID": {
+          "type": "integer",
+          "description": "Group ID to squash to.",
+          "format": "int64"
+        },
+        "status": {
+          "readOnly": true,
```

Comment:
can the status be enum in this case?

Reply:
This is an unstructured text string that can change from release to release, but the customer(s) will be familiar with the output from Lustre (Filesystem).